StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/UpdatePaymentMethodViewController.swift

```diff
@@ -235,14 +235,31 @@ final class UpdatePaymentMethodViewController: UIViewController {
         let updatePaymentMethodResult = await delegate.didUpdate(viewController: self, paymentMethod: configuration.paymentMethod)
         switch updatePaymentMethodResult {
         case .success:
-            STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: configuration.hostedSurface.analyticEvent(for: .updateCard),
-                                                                 params: analyticsParams)
+            if case .card(let paymentMethodCardParams) = updatePaymentMethodOptions,
+               let selectedCardBrand = paymentMethodCardParams.networks?.preferred {
+                STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: configuration.hostedSurface.analyticEvent(for: .updateCard),
+                                                                     params: ["selected_card_brand": selectedCardBrand])
+            }
+            if setAsDefaultValue == true {
+                STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: configuration.hostedSurface.analyticEvent(for: .setDefaultPaymentMethod),
+                                                                     params: ["payment_method_type": configuration.paymentMethod.type.identifier])
+            }
         case .failure(let errors):
             updateButton.update(state: .enabled)
             latestError = errors.count == 1 ? errors[0] : NSError.stp_genericErrorOccurredError()
-            STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: configuration.hostedSurface.analyticEvent(for: .updateCardFailed),
-                                                                 error: latestError,
-                                                                 params: analyticsParams)
+            if errors.contains(where: { error in error as NSError == NSError.stp_cardBrandNotUpdatedError() }) {
```

Comment:
Super super nit, can we put parenthesis around the casting, I think it will read easier. Same w/ below.
`error in (error as NSError) == NSError.stp_cardBrandNotUpdatedError()`

Another super nit: I find it hard to read b/c there are so many instances of the same word "error", it is a little hard for me to follow. An alternative way to write this is:

`if errors.contains(where: { ($0 as NSError) == NSError.stp_cardBrandNotUpdatedError() })`